timesketch/lib/analyzers/yetiindicators.py

```diff
@@ -241,33 +294,71 @@ def add_intelligence_entry(self, indicator, event, entity):
         intel = {
             "externalURI": uri,
             "ioc": match_in_sketch,
-            "tags": indicator["relevant_tags"] + [entity["name"]],
-            "type": "other",
+            "tags": [slugify(tag) for tag in tags + [entity["name"]]],
+            "type": intel_type,
         }
 
         self._intelligence_attribute["data"].append(intel)
         self._intelligence_refs.add((match_in_sketch, uri))
 
-    def get_intelligence_attribute(self):
+    def get_intelligence_attribute(self) -> Tuple[Dict, Set[Tuple[str, str]]]:
+        """Fetches the intelligence attribute from the database."""
         try:
-            self._intelligence_attribute = self.sketch.get_sketch_attributes(
-                "intelligence"
-            )
-            self._intelligence_refs = {
+            intelligence_attribute = self.sketch.get_sketch_attributes("intelligence")
+            refs = {
                 (ioc["ioc"], ioc["externalURI"])
-                for ioc in self._intelligence_attribute["data"]
+                for ioc in intelligence_attribute["data"]
             }
+            return intelligence_attribute, refs
+
         except ValueError:
-            print("Intelligence not set on sketch, will be created from scratch.")
+            logging.info(
+                "Intelligence not set on sketch, will be created from scratch."
+            )
+        return {"data": []}, set()
+
+    def save_intelligence(self) -> None:
```

Comment:
Would it be beneficial for the analyzer to have an overview of added intelligence in the analyzer results? (similar to tags?)

Reply:
Yes, that would be a good idea! Maybe the number of new elements added would be enough.

Reply:
This should be handled by the function requested in #3082